pkg/kubelet/cm/cpumanager/topology_hints.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cpumanager
+
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/klog"
+
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/topology"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/socketmask"
+)
+
+func (m *manager) GetTopologyHints(pod v1.Pod, container v1.Container) []topologymanager.TopologyHint {
+	// The 'none' policy does not generate topology hints.
+	if m.policy.Name() == string(PolicyNone) {
+		return nil
+	}
+
+	// For all other policies, if there are no CPU resources requested for this
+	// container, we do not generate any topology hints.
+	if _, ok := container.Resources.Requests[v1.ResourceCPU]; !ok {
+		return nil
+	}
+
+	// Otherwise, attempt to generate TopologyHints for the CPUManager.
+	// For now, this implementation assumes the 'static' CPUManager policy.
+	// TODO: Generalize this so that its applicable to future CPUManager polices.
```

Comment:
Can we make an issue to follow up with this?

Reply:
`future CPUManager polices` does it mean a kind of thread sibling policies?

Reply:
It’s not talking about any particular new policy. The main point here is that this function implementation breaks the policy abstraction we have set up. It would be preferable to add a new method to the Policy interface and then delegate to that from here instead.